src/sidecar/connections/watcher.ts

```diff
@@ -41,17 +47,42 @@ export async function waitForConnectionToBeStable(
     // and trigger any kind of Topics/Schemas refresh to reenforce the error state / clear out any
     // old data
     environmentChanged.fire(id as unknown as EnvironmentId);
+    const lastConnectionEvent = connectionStateWatcher.getLatestConnectionEvent(id);
+    if (lastConnectionEvent) {
+      showErrorNotificationWithButtons(
+        `Timed out establishing "${lastConnectionEvent.connection.spec.name}" connection.`,
+      );
+    }
     logger.error(msg);
     return null;
   }
 
   logger.debug(
     `waitForConnectionToBeStable(): connection is stable after ${Date.now() - startTime}ms, returning`,
   );
+  notifyForFailedState(connection);
 
   return connection;
 }
 
+function notifyForFailedState(connection: Connection) {
```

Comment:
```suggestion
function notifyIfailedState(connection: Connection) {
```

Rename -- it only notifies IF there is a failed state.

Reply:
Done in https://github.com/confluentinc/vscode/pull/946/commits/608e3b72c005703002e57f01681a7be15710e324